kernel/arch/dreamcast/include/dc/pvr.h

```diff
@@ -17,1773 +18,206 @@
-    \ingroup                        pvr_pal_mgmt
-
-    Entries in the PVR's palettes can be of any of these formats. Note that you
-    can only have one format active at a time.
-
-    @{
-*/
-typedef enum pvr_palfmt {
-    PVR_PAL_ARGB1555,        /**< \brief 16-bit ARGB1555 palette format */
-    PVR_PAL_RGB565,          /**< \brief 16-bit RGB565 palette format */
-    PVR_PAL_ARGB4444,        /**< \brief 16-bit ARGB4444 palette format */
-    PVR_PAL_ARGB8888,        /**< \brief 32-bit ARGB8888 palette format */
-} pvr_palfmt_t;
-/** @} */
-
-/** \brief   Set the palette format.
-    \ingroup pvr_pal_mgmt
-
-    This function sets the currently active palette format on the PVR. Each
-    entry in the palette table is 32-bits in length, regardless of what color
-    format is in use.
-
-    Be sure to use care when using the PVR_PAL_ARGB8888 format. Rendering speed
-    is greatly affected (cut about in half) if you use any filtering with
-    paletted textures with ARGB8888 entries in the palette.
-
-    \param  fmt             The format to use
-    \see    pvr_palfmt_t
-*/
-void pvr_set_pal_format(pvr_palfmt_t fmt);
-
-/** \brief   Set a palette value.
-    \ingroup pvr_pal_mgmt
-
-    Note that while the color format is variable, each entry is still 32-bits in
-    length regardless (and you only get a total of 1024 of them). If using one
-    of the 16-bit palette formats, only the low-order 16-bits of the entry are
-    valid, and the high bits should be filled in with 0.
-
-    \param  idx             The index to set to (0-1023)
-    \param  value           The color value to set in that palette entry
-*/
-static inline void pvr_set_pal_entry(uint32_t idx, uint32_t value) {
-    PVR_SET(PVR_PALETTE_TABLE_BASE + 4 * idx, value);
-}
-
-
-/* Hardware Fog parameters *******************************************/
-/** \defgroup   pvr_fog     Fog
-    \brief                  Hardware Fog API for the PowerVR
-    \ingroup                pvr_global
-
-    \note 
-    Thanks to Paul Boese for figuring this stuff out
-*/
-
-/** \brief   Set the table fog color.
-    \ingroup pvr_fog
-
-    This function sets the color of fog for table fog. 0-1 range for all colors.
-
-    \param  a               Alpha value of the fog
-    \param  r               Red value of the fog
-    \param  g               Green value of the fog
-    \param  b               Blue value of the fog
-*/
-void pvr_fog_table_color(float a, float r, float g, float b);
-
-/** \brief   Set the vertex fog color.
-    \ingroup pvr_fog
+    \remark
+    This file is used for pretty much everything related to the PVR, from memory
+    management to actual primitive rendering.
 
-    This function sets the fog color for vertex fog. 0-1 range for all colors.
-    This function is currently not implemented, as vertex fog is not supported
-    by KOS. Calling this function will cause an assertion failure.
+    \todo 
+        - Auxiliary accumulation buffer
```

Comment:
That is already supported.

Reply:
You're right. At the time I had no idea how it was actually used, and everyone told me it wasn't something they knew how to use either... just need probably better docs and/or example.